gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/analytics/adapter/SearchTopHitsAnalyticsByEntrypointResponseAdapter.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.repository.elasticsearch.v4.analytics.adapter;
+
+import static io.gravitee.repository.elasticsearch.v4.analytics.adapter.SearchTopHitsAnalyticsByEntrypointQueryAdapter.ENTRYPOINT_ID_AGG;
+import static io.gravitee.repository.elasticsearch.v4.analytics.adapter.SearchTopHitsAnalyticsByEntrypointQueryAdapter.STATUS_RANGES;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.gravitee.elasticsearch.model.Aggregation;
+import io.gravitee.elasticsearch.model.SearchResponse;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class SearchTopHitsAnalyticsByEntrypointResponseAdapter {
+
+    public static Optional<Map<String, Map<String, Long>>> adapt(SearchResponse response) {
```

Comment:
Wouldn't it be possible to have the same as for the other 3 analytics (CountAggregate & AverageAggregate )? 
something like 
```
public class TopHitsAggregate {

    Map<String, Long> topHits; // For all entrypoints
    Map<String, Map<String, Long>> topHitsBy; // for each entypoints
}
```